salesforce-composite/2.7/modules/ROOT/pages/edits/salesforce-composite-connector-examples.adoc

```diff
@@ -0,0 +1,469 @@
+   http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+http://www.mulesoft.org/schema/mule/http
+http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+http://www.mulesoft.org/schema/mule/ee/core
+http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd
+http://www.mulesoft.org/schema/mule/salesforce-composite
+http://www.mulesoft.org/schema/mule/salesforce-composite/current/mule-salesforce-composite.xsd">
+	<configuration-properties file="mule-app.properties"/>
+	<http:listener-config name="HTTP_Listener_config"
+   doc:name="HTTP Listener config">
+		<http:listener-connection host="localhost" port="8081" />
+	</http:listener-config>
+	<salesforce-composite:composite-config name="Salesforce_Composite_Salesforce_Composite"
+   doc:name="Salesforce Composite Salesforce Composite">
+		<salesforce-composite:oauth-user-pass-connection
+      consumerKey="${consumerKey}"
+      consumerSecret="${consumerSecret}"
+      username="${username}"
+      password="${password}"
+      securityToken="${securityToken}"
+      tokenEndpoint="${tokenEndpoint}" />
+	</salesforce-composite:composite-config>
+	<flow name="salesforce-composite-create-sobject-trees">
+		<http:listener doc:name="Listener"
+      path="/createSObjectTree"
+      config-ref="HTTP_Listener_config"/>
+		<ee:transform doc:name="Transform HTTP input for Create SObject Tree">
+			<ee:message>
+				<ee:set-payload><![CDATA[%dw 2.0
+output application/java
+---
+payload.records map ((record , indexOfRecord) -> {
+	"attributes": {
+		"type": record.attributes.objectType,
+		"referenceId": record.attributes.referenceId
+	},
+	"Name": record.Name,
+	"Phone": record.Phone,
+	"Website": record.Website,
+	"NumberOfEmployees": record.NumberOfEmployees as Number,
+	("ChildAccounts": {
+		"records": record.ChildAccounts.records map ((record01, indexOfRecord01) -> {
+			"attributes": {
+				"type": record01.attributes.objectType,
+				"referenceId": record01.attributes.referenceId
+			},
+			"Name": record01.Name,
+			"Phone": record01.Phone,
+			"Website": record01.Website,
+			"NumberOfEmployees": record01.NumberOfEmployees as Number
+		})
+	}),
+	"Contacts": {
+		"records": record.Contacts.records map ((record01, indexOfRecord01) -> {
+			"attributes": {
+				"type": record01.attributes.objectType,
+				"referenceId": record01.attributes.referenceId
+			},
+			"LastName": record01.LastName,
+			"Email": record01.Email,
+			"Title": record01.Title
+		})
+	}
+})
+]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+		<salesforce-composite:create-sobject-tree
+      doc:name="Create SObject tree"
+      config-ref="Salesforce_Composite_Salesforce_Composite"
+      type="Account"/>
+		<ee:transform doc:name="Transform Message">
+			<ee:message >
+				<ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload]]></ee:set-payload>
+			</ee:message>
+		</ee:transform>
+	</flow>
+</mule>
+----
+
+== Pre-Query Example
```

Comment:
pre-query or pre-create?
When use as the operation name, Pre-create or Pre-Create?
GLOBAL